lib/locker/advisory.rb

```diff
@@ -32,8 +34,13 @@ def self.run(key, options={}, &block)
     def run(&block)
       connection = ActiveRecord::Base.connection_pool.checkout
       connection.transaction :requires_new => true do
+        if @blocking && @block_timeout
+          break_at = Time.now + @block_timeout
+        end
+
         while !get(connection) && @blocking
-          sleep 0.005
+          sleep @block_spin_wait
+          break if break_at && break_at < Time.now
```

Comment:
I was thinking that with `@block_spin_wait` it's more likely to be past the timeout after a sleep (assuming the `get` is shorter than 5ms), and so we should check then. It's probably better to check before and after the sleep, really, but if we just do the one then I don't care much which side of the sleep it goes on.